internal/hardware/validator.go

```diff
@@ -150,9 +154,60 @@ func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infra
 		}
 	}
 
+	if disk.DriveType == models.DriveTypeISCSI {
+		err := isISCSINetworkingValid(disk, inventory)
+		if err != nil {
+			notEligibleReasons = append(notEligibleReasons, err.Error())
+		}
+	}
+
 	return notEligibleReasons, nil
 }
 
+// isISCSINetworkingValid checks if the iSCSI dish is not connected through the
+// default network interface. The default network interface is the interface
+// which is used by the default gateway.
+func isISCSINetworkingValid(disk *models.Disk, inventory *models.Inventory) error {
+	// get the IPv4 or the IPv6 of the interface connected to the iSCSI target
+	if disk.Iscsi == nil {
+		return fmt.Errorf("Host IP address is not available")
+
+	}
+	iSCSIHostIP := net.ParseIP(disk.Iscsi.HostIPAddress)
+	if iSCSIHostIP == nil {
+		return fmt.Errorf("Cannot parse iSCSI host IP %s", disk.Iscsi.HostIPAddress)
+	}
+	isIPv6 := govalidator.IsIPv6(iSCSIHostIP.String())
+
+	defaultRoute := network.GetDefaultRouteByFamily(inventory.Routes, isIPv6)
+	if defaultRoute == nil {
```

Comment:
basically, no default route means internal network? Or is there something else to check?

Reply:
If we reached this point then there is probably a  route.
Anyway, I don't think you should abort in case no default route is found because you don't require it.

Reply:
done! I return `nil` now